### PR TITLE
User image upload page

### PR DESCRIPTION
**Pull request by Brechy**
There is a lot going on in here but it is working smoothly. I had to go in and change some more code in the signup form to make it work properly and fixed some bugs in the signup form code. Added gravatar default image service (puts gravatar image url into database from signup page), stored the user's email address in local storage to accommodate the requirement of remember the user as we go through the signup process. Please take a look, there will be a lot of merge conflicts I am sure but it might be worth it to go with them as they solve a lot of issues we were needing to sort out anyway. 🐧

### CODE REVIEW

**Comment by Brechy** [comment]
Corrected all merge conflicts, using new code created in this branch in signup files.

File: public/user_img.html [hunk 1]
```diff
@@ -32,7 +32,7 @@
     <!-- Navigation -->
     <nav class="navbar navbar-expand-lg bg-secondary fixed-top text-uppercase" id="mainNav">
       <div class="container">
-        <a class="navbar-brand js-scroll-trigger" href="#page-top">Fam</a>
+        <a href="index.html" class="navbar-brand js-scroll-trigger" href="#page-top">Fam</a>
```
Comment (JBallin) [inline public/user_img.html]: not sure about multiple href's here, personally I would have one href pointing to "/"

File: public/user_img.html [hunk 2]
```diff
@@ -42,13 +42,13 @@
         <div class="collapse navbar-collapse" id="navbarResponsive">
           <ul class="navbar-nav ml-auto">
             <li class="nav-item mx-0 mx-lg-1">
-              <a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#portfolio">Home</a>
+              <a href="index.html" class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger">Home</a>
```
Comment (JBallin) [inline public/user_img.html]: would have href point to "/" so that it doesn't show up as index.html in the URL

File: public/user_img.html [hunk 3]
```diff
@@ -42,13 +42,13 @@
...
             </li>
             <li class="nav-item mx-0 mx-lg-1">
-              <a class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger" href="#about">LOGIN</a>
+              <a href="login.html" class="nav-link py-3 px-0 px-lg-3 rounded js-scroll-trigger">LOGIN</a>
```
Comment (JBallin) [inline public/user_img.html]: In #41 I proposed keeping this as "/signup" for now but I'm happy to have it link to a "/login" route that sends back `login.html`. Current is fine though I suppose

File: public/javascripts/signupform.js [hunk 4]
```diff
@@ -1,36 +1,58 @@
...
+const userSignup = () => {
+	try {
+		const email =  document.querySelector('#email').value;
+		const password = document.querySelector('#password').value;
+		const name = document.querySelector('#name').value;
+		console.log('in user signup', email, password, name);
 
-    let response = await data.json()
-    return response
+		fetch('http://localhost:3000/signup', {
```
Comment (JBallin) [inline public/javascripts/signupform.js]: Should be `fetch('/signup')`

File: public/user_img.html (hunk 1 above)
Comment (JBallin) [reply public/user_img.html]: Also, see https://github.com/Brechy/fam/issues/47

**Comment by JBallin** [comment]
@Brechy is working on this. She'll re-open when ready.

**Pull request closed without merging**